build.gradle

```diff
@@ -80,39 +79,21 @@ define default value for the rust binary build type
 ext.rustBinaryBuildType = project.findProperty('rustBinaryBuildType') ?: 'debug'
 
 /*
- * Creates a nice version name we can use for built assets; from the git version.
+ * Get the version name from an environment variable (or use the default). Used for built assets;
  * This is set to gradle's version property below.
  */
 def radixVersion() {
-    /*
-     KNOWN ISSUE:
-     1. If there are two (simple) tags on the same commit, it selects the oldest.  This can be a problem if (eg) a
-        release and a release candidate tag share the same commit.
-
-        EXPLANATION:
-        versionDetails is from https://github.com/palantir/gradle-git-version
-        Behind the scenes, it parses "git describe --tags --always --first-parent"
-        This is used to extract the closest tag information. If there are multiple, git describe attempts to select
-         the most recent tag - but this only works if the tags are annotated - if they're simple tags, it just
-         selects the first.
-        Github release creates simple tags, not annotated tags (as of April 2022) - see discussion here
-         for suggested change: https://github.com/github/feedback/discussions/4924
-
-        WORKAROUND:
-          Manually create the annotated tag yourself for the release, eg:
-            git tag -a v1.2.1 -m "Version 1.2.1"
-          See also: https://git-scm.com/book/en/v2/Git-Basics-Tagging
-    */
-    def details = versionDetails()
-
+    def versionBranchEnvVar = System.getenv('VERSION_BRANCH')
+    def versionBuildEnvVar = System.getenv('VERSION_BUILD')
+    def versionTagEnvVar = System.getenv('VERSION_TAG')
     def version
-    if (details.isCleanTag) {
-        version = details.lastTag
+    if (versionTagEnvVar && !versionTagEnvVar.allWhitespace) {
+        version = versionTagEnvVar
     } else {
-        if (details.branchName == null) {
-            version = "detached-head-${details.gitHash}"
+        if (versionBranchEnvVar == null) {
+            version = "detached-head-$versionTagEnvVar"
         } else {
-            version = "${details.branchName}-${details.gitHash}"
+            version = "$versionBranchEnvVar-$versionBuildEnvVar"
         }
         // Prepend SNAPSHOT so the tag name is valid regardless of the version
         version = "SNAPSHOT-${version}"
```

Comment:
If `version` is empty (e.g. locally) we probably want to not use `SNAPSHOT`. Perhaps `SNAPSHOT-development` or something might be okay?

Reply:
I'm not actually sure what the impact might be locally if we end up building the same version multiple times? Possibly not a big deal, or it might break some build things... I guess we'll find out.

Reply:
Updated to use development, if the `versionBranchEnvVar` is empty